src/main/java/org/folio/rest/support/HridManager.java

```diff
@@ -0,0 +1,233 @@
+                    conn, existingHridSettings, hridSettings))
+                .map(success -> endTransaction(conn, promise))
+                .otherwise(error -> rollback(conn, promise, error));
+            } catch (Exception e) {
+              fail(promise, "Failed to update HRID settings", e);
+            }
+          });
+        } catch (Exception e) {
+          fail(promise, "Failed creating a database transaction", e);
+        }
+      });
+    } catch (Exception e) {
+      fail(promise, "Failed to execute updating the HRID settings in a context", e);
+    }
+
+    return promise.future();
+  }
+
+  private Future<HridSettings> getHridSettings(AsyncResult<SQLConnection> conn) {
+    final String sql = "SELECT jsonb FROM " + HRID_SETTINGS_TABLE;
+    final Promise<JsonArray> promise = Promise.promise();
+
+    try {
+      postgresClient.selectSingle(conn, sql, promise);
+    } catch (Exception e) {
+      fail(promise, "Failed to get HRID settings from the database", e);
+    }
+
+    return promise.future().map(
+        results -> Json.decodeValue(results.getString(0), HridSettings.class));
+  }
+
+  private Future<Void> updateHridSettings(AsyncResult<SQLConnection> conn,
+      HridSettings existingHridSettings, HridSettings hridSettings) {
+    hridSettings.setId(existingHridSettings.getId());
+
+    final Promise<UpdateResult> promise = Promise.promise();
+
+    try {
+      postgresClient.update(conn, HRID_SETTINGS_TABLE, hridSettings, null, false, promise);
+
+      // Even though we are executing this within a transaction, all updates must be done
+      // sequentially or else the SQL client will report race condition errors due to a
+      // currently running query. The order is arbitrary, but first update the settings, then
+      // update the instance sequence, then update the holdings sequence and finally update
+      // the item sequence.
+      return promise.future().compose(v -> updateSequence(conn, "instance",
+          existingHridSettings.getInstance(), hridSettings.getInstance())
+          .compose(v1 -> updateSequence(conn, "holding", existingHridSettings.getHolding(),
+                  hridSettings.getHolding())
+              .compose(v2 -> updateSequence(conn, "item", existingHridSettings.getItem(),
+                      hridSettings.getItem()))));
+    } catch (Exception e) {
+      fail(promise, "Failed to update the HRID settings and sequences", e);
+    }
+
+    return promise.future().map(v -> null);
+  }
+
+  private Future<Void> updateSequence(AsyncResult<SQLConnection> conn, String field,
+      HridSetting existingHridSetting, HridSetting hridSetting) {
+    final Promise<ResultSet> promise = Promise.promise();
+
+    // Only update the sequence if the start number has changed
+    if (existingHridSetting.getStartNumber().intValue() !=
+        hridSetting.getStartNumber().intValue()) {
+      final String sql = String.format("select setval('hrid_%s_seq',%d,FALSE)",
```

Comment:
Maybe more requirement question. As far as I remember the HRIDs should be unique per items/instances/holdings, am I correct?

If so, how should we handle following case then:
1. Let's say the start number is **6** now;
1. We've issued some hrids, so the current val for a sequence is 9 how.
1. then we modifying start number to either **1-8** now (it is less then current value of the sequence);
1. this method updates the sequences, so the current `next val` becomes less then previous `next val`;
1. when we call the sequence to get a next value, the resulting hrid will be duplicating already issued one.

Reply:
@bohdan-suprun I agree with this concern. I asked a similar question during refinement I believe, and IIRC the requirements position was that staff _should not_ change the configuration of the working system. Yet, we aren't going to stop them from doing so, the systems behaviour after which is regarded as unsupported.

@mreno-EBSCO Do you have a similar recollection of the requirements conversation?

Maybe this needs stating explicitly in the JIRA issue so that folks understand this limitation was by design?

Reply:
You are correct. That is my understanding as well. The tenant needs to know that resetting the start number is not recommended unless they change the prefix. If they do this, they won't be able save the record since there is a unique index on the hrid. This is something a technical manual should detail with some sort of reference in the UI. I'm not sure how apparent we can make this in the API docs. I'll think about that and see if I can come up with something.

Reply:
@mreno-EBSCO I've asked this question for confirmation on the issue. I think we should add it to the module readme as a starting point.

(I'm aware the readme isn't very discoverable, and has the challenge of interface vs. implementation)

Reply:
@marcjohnson-kint OK, I'll put something in the README and add a note in the JIRA for PO visibility. I am pretty sure this is the expected behavior. Trying to manage this level of bookkeeping would be challenging, doable, but challenging. Likewise, if the set the start number to 99999999, they get exactly 1 hrid before the sequence is maxed out.

Another thing that just occurred to me is, if the prefix is changed, but not the start number, then the sequence is not reset to the start start number. I am not sure if that is the correct behavior. Hmm...

Reply:
I agree, there a number of different scenarios around this that could cause issues.

This is an example of settings that should only be set in an initial setup period, and then locked (it's not even write once, because they have defaults). 

However FOLIO has no explicit modelling of this concept, which I think is what causes this.

Reply:
I think for the moment, the most we can do is issue a warning in the documentation for the module (and the UI maybe).

Reply:
Sorry, it's not clear for me how we can change the start value for the sequence on a fly using API?

Reply:
> Sorry, it's not clear for me, but how we can change the start value for the sequence on a fly using API?

@OleksiiKuzminov, that's the whole purpose of this PR. Are you unclear on how it is implemented or whether or not this should even be possible?